src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -1070,8 +1081,11 @@ def halt(self) -> None:
         vmss = get_vmss(self.scaleset_id)
         if vmss:
             logging.info("scaleset deleting: %s", self.scaleset_id)
-            delete_vmss(self.scaleset_id)
-            self.save()
+            if delete_vmss(self.scaleset_id):
+                logging.info("scaleset deleted: %s", self.scaleset_id)
```

Comment:
Now, `delete_vmss` returns `True` if there is a `CloudError` or `ResourceError`, and internally logs at error level.

1. Doesn't that make this `info` log message incorrect?
2. Why do we `delete()`? That makes sense for a `ResourceNotFoundError`, but it seems like we also can reach this branch due to a `CloudError`, in which case the underlying VMSS may still exist, but we'd delete our `Scaleset` metadata which tracks it, leaving the Azure resource dangling. Am I misreading?

Reply:
Split the exceptions in half.  If it's ResourceNotFoundError, then treat it as it's already deleted, which was my original intent.

Reply:
I think the business logic now correct, but we still want to update the logging here.

How about in the `True` branch, we log something like `"underlying scaleset resource already deleted"`, and `"scaleset deleted"` in the other branch (since that's the case where we actually performed a deletion)?

Reply:
Having rethought this, we can just always attempt the delete, rather than `get` then `delete`, saving a round trip to Azure.

If delete_vmss returns true, it's deleted.  otherwise, we save the scaleset which will cause the delete to get retried alter on.

Reply:
Great. I'd only also suggest you note the expectation that `save()` assumes "we'll retry deletion failed", since that's important but not locally obvious in the source code.